tests/d3d12_shaders.c

```diff
@@ -14732,3 +14732,156 @@ void test_memory_model_uav_coherence_thread_group_dxil(void)
+        0x66, 0x40, 0x0d, 0x97, 0xef, 0x3c, 0x7e, 0x40, 0x15, 0x05, 0x11, 0x95, 0x0e, 0x30, 0xf8, 0xc8, 0x6d, 0x5b, 0x41, 0x35, 0x5c, 0xbe, 0xf3, 0xf8, 0x01, 0x55, 0x14, 0x44, 0xc4, 0x4e, 0x4e, 0x44,
+        0xf8, 0xc8, 0x6d, 0x1b, 0x81, 0x34, 0x5c, 0xbe, 0xf3, 0xf8, 0x42, 0x44, 0x00, 0x13, 0x11, 0x02, 0xcd, 0xb0, 0x10, 0x26, 0x40, 0x0d, 0x97, 0xef, 0x3c, 0x3e, 0x44, 0x01, 0x13, 0x11, 0x35, 0x40,
+        0x84, 0xf9, 0xc8, 0x6d, 0xdb, 0x80, 0x34, 0x5c, 0xbe, 0xf3, 0xf8, 0x13, 0x11, 0x4d, 0x08, 0x10, 0x61, 0x3e, 0x72, 0xdb, 0x16, 0x20, 0x0d, 0x97, 0xef, 0x3c, 0xfe, 0x74, 0x44, 0x04, 0x30, 0x88,
+        0x83, 0x8f, 0xdc, 0xb6, 0x01, 0x10, 0x0c, 0x80, 0x34, 0x00, 0x00, 0x00, 0x61, 0x20, 0x00, 0x00, 0x33, 0x00, 0x00, 0x00, 0x13, 0x04, 0x41, 0x2c, 0x10, 0x00, 0x00, 0x00, 0x06, 0x00, 0x00, 0x00,
+        0x34, 0x8a, 0xac, 0x20, 0x4a, 0xad, 0x60, 0x4a, 0xae, 0x74, 0x03, 0x0a, 0x31, 0xa0, 0x14, 0x03, 0x48, 0x95, 0xc0, 0x08, 0x40, 0x79, 0x00, 0x00, 0x23, 0x06, 0x09, 0x00, 0x82, 0x60, 0xe0, 0x64,
+        0xc7, 0x50, 0x55, 0xcf, 0x88, 0x41, 0x02, 0x80, 0x20, 0x18, 0x38, 0x1a, 0x32, 0x58, 0x16, 0x34, 0x62, 0x60, 0x00, 0x20, 0x08, 0x06, 0x84, 0x87, 0x5c, 0x23, 0x06, 0x08, 0x00, 0x82, 0x60, 0x30,
+        0x69, 0x88, 0x10, 0x60, 0xa3, 0x09, 0xc1, 0x30, 0x9a, 0x20, 0x04, 0xa3, 0x09, 0x83, 0x30, 0x9a, 0x40, 0x00, 0x35, 0x48, 0x38, 0x62, 0x90, 0x00, 0x20, 0x08, 0x06, 0x06, 0x19, 0x48, 0x53, 0x34,
+        0x04, 0x55, 0x6c, 0x38, 0x62, 0x90, 0x00, 0x20, 0x08, 0x06, 0x86, 0x19, 0x50, 0x1c, 0x57, 0x04, 0x75, 0x50, 0x38, 0x62, 0x90, 0x00, 0x20, 0x08, 0x06, 0x06, 0x1a, 0x58, 0xd5, 0x75, 0x04, 0x23,
+        0x06, 0x08, 0x00, 0x82, 0x60, 0x80, 0xa0, 0x41, 0x76, 0x24, 0xca, 0x88, 0x01, 0x02, 0x80, 0x20, 0x18, 0x20, 0x69, 0xa0, 0x21, 0xc6, 0x32, 0x62, 0x80, 0x00, 0x20, 0x08, 0x06, 0x88, 0x1a, 0x6c,
+        0x49, 0xc1, 0x8c, 0x18, 0x20, 0x00, 0x08, 0x82, 0x01, 0xb2, 0x06, 0x9c, 0x42, 0x34, 0x23, 0x06, 0x0d, 0x00, 0x82, 0x60, 0x00, 0xa9, 0x81, 0x25, 0x41, 0x67, 0x40, 0x0c, 0x42, 0x30, 0x21, 0x00,
+        0x00, 0x00, 0x00, 0x00,
+    };
+    static const D3D12_SHADER_BYTECODE cs_code_dxil = SHADER_BYTECODE(cs_code);
+
+    /* Accumulation above 16-bit is supposed to be undefined, but we'll have to test what really happens. */
+    static const struct input
+    {
+        uint8_t reference[4];
+        uint8_t source[8];
+        uint32_t accum;
+    } input_data[] = {
+        { { 0, 1, 2, 3 }, { 4, 5, 6, 7, 8, 9, 10, 11 }, 10 },
+        { { 0, 1, 2, 3 }, { 4, 5, 6, 7, 8, 9, 10, 11 }, 0xffffffffu },
+        { { 0, 0, 0, 0 }, { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff }, 123 },
+        { { 0xff, 0xff, 0xff, 0xff }, { 0, 0, 0, 0, 0, 0, 0, 0 }, 123 },
+        { { 0xff, 0xff, 0xff, 0xff }, { 1, 2, 3, 0x80, 6, 7, 8, 9 }, 8000 },
+        { { 0xab, 0x13, 0x25, 0xbc }, { 0xaa, 0x2, 0x7, 0x9, 0x30, 0x50, 0x80, 0x234 }, 0x1234 },
```

Comment:
> 0x234

This seems broken since this is still part of the 8-bit input array.

Reply:
Fixed.